plaso/output/elastic.py

```diff
@@ -13,7 +13,70 @@ class ElasticsearchOutputModule(shared_elastic.SharedElasticsearchOutputModule):
 
   NAME = 'elastic'
   DESCRIPTION = 'Saves the events into an Elasticsearch database.'
-
+  DEFAULT_MAPPING = {
+      'properties': {
+          "application": {
+              "type": "text",
+              "fields": {
+                  "keyword": {"type": "keyword"}},
+          },
+          "data": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "doc_type": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "event_type": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "exit_status": {"type": "text",
+                          "fields": {"keyword": {"type": "keyword"}},
+                         },
+          "facility": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "file_reference": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "file_size": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "flags": {"type": "text", "fields": {"keyword": {"type": "keyword"}}},
+          "identifier": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "message_status": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "message_type": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "offset": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "sequence_number": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "severity": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+          "source_port": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "user_identifier": {
+              "type": "text",
+              "fields": {"keyword": {"type": "keyword"}},
+          },
+          "version": {
+              "type": "text", "fields": {"keyword": {"type": "keyword"}},
+          },
+      }
+  }
```

Comment:
I'm wondering... how often will this list change? is it possible to create this more robustly, what happens when a new parser/plugin comes in that has the same attribute name as someone else, then we need to have a way to check whether or not we need to change this mapping, so I'm wondering whether we can have at least some sort of maintenance script that can do the check and add to this list?

Reply:
I have already made a script  to detect these collision. I have no longer access to it but can recreate it/ ask to get it back. this script basically parse the comment from parser and extract the field name and type for each data type. Once every parser have been parsed, it detect if some field with the same name have different type.

It's not perfect because the comment could be inacurrate (E.g for the windows:srum:application_usage data_type, the application field is specified as a string, but in some case it could also be an integer) 

I don't think this list would change really often, because most of time if field have the same name, they contains the same type of data

Reply:
ok, we should probably think about having a script of some sort checked in as well, to be able to maintain this.